packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -80,6 +80,78 @@ void main() {
     debugDefaultTargetPlatformOverride = null;
   });
 
+  testWidgets('Using other pickers that rebuild the switch will not cause vibrations', (WidgetTester tester) async {
+    debugDefaultTargetPlatformOverride = TargetPlatform.iOS;
+    final Key switchKey = UniqueKey();
+    final Key switchKey2 = UniqueKey();
+    bool value = false;
+    bool value2 = false;
+    final List<MethodCall> log = <MethodCall>[];
+
+    SystemChannels.platform.setMockMethodCallHandler((MethodCall methodCall) async {
+      log.add(methodCall);
+    });
+
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return Center(
+              child: Column(
+                children: <Widget>[
+                  CupertinoSwitch(
+                    key: switchKey,
+                    value: value,
+                    onChanged: (bool newValue) {
+                      setState(() {
+                        value = newValue;
+                      });
+                    },
+                  ),
+                  CupertinoSwitch(
+                    key: switchKey2,
+                    value: value2,
+                    onChanged: (bool newValue) {
+                      setState(() {
+                        value2 = newValue;
+                      });
+                    },
+                  ),
+                ]
+              ),
+            );
+          },
+        ),
+      ),
+    );
+
+    await tester.tap(find.byKey(switchKey));
+    await tester.pumpAndSettle();
+
+    expect(log, hasLength(1));
+    expect(log[0], isMethodCall('HapticFeedback.vibrate', arguments: 'HapticFeedbackType.lightImpact'));
+
+    await tester.tap(find.byKey(switchKey2));
+    await tester.pumpAndSettle();
+
+    expect(log, hasLength(2));
+    expect(log[1], isMethodCall('HapticFeedback.vibrate', arguments: 'HapticFeedbackType.lightImpact'));
+
+    await tester.tap(find.byKey(switchKey));
+    await tester.pumpAndSettle();
+
+    expect(log, hasLength(3));
+    expect(log[2], isMethodCall('HapticFeedback.vibrate', arguments: 'HapticFeedbackType.lightImpact'));
+
+    await tester.tap(find.byKey(switchKey2));
+    await tester.pumpAndSettle();
+
+    expect(log, hasLength(4));
+    expect(log[3], isMethodCall('HapticFeedback.vibrate', arguments: 'HapticFeedbackType.lightImpact'));
+    debugDefaultTargetPlatformOverride = null;
+  });
+
   testWidgets('Switch can drag (LTR)', (WidgetTester tester) async {
```

Comment:
Can you add another test checking that dragging the knob (instead of just tapping it) to the other edge also does whatever haptic feedback iOS does?